proposals/0002-explicit-fixity-(~).md

```diff
@@ -0,0 +1,55 @@
+---
+author: Rinat Stryungis
+date-accepted: ""
+ticket-url: "https://gitlab.haskell.org/ghc/ghc/-/issues/18252"
+implemented: "Yes"
+---
+
+This proposal is [discussed at this pull request](https://github.com/haskell-core/core-libraries-proposals/pull/0>).
+**After creating the pull request, edit this file again, update the number in
+the link, and delete this bold sentence.**
+
+# Proposal title
+
+Added explicit fixity for `(~)` and `(~~)`
+
+## Motivation
+
+It solves the issue by adding an explicit fixity for the `(~)` type operator.
+It has been set equal to 4:
+
+    infix 4 ~, ~~
+
+And there are two reasons for that:
+
+1. These fixities are equal to the corresponding fixities of `(:~:)` and `(:~~:)`
+2. They are less than casual arithmetic operators and more type-level arithmetic expressions could be written without brackets.
+
+## Proposed Change Specification
+
+1. Type operator `(~)` is now explicitly non-associative with fixity 4 
+2. Type operator `(~~)` is now explicitly non-associative with fixity 4 
+
+## Examples
+
+1. `a ~ b ~ c` now would be rejected with an error message about fixity, not with a type error.
+2. Expression `5 + 5 ~ 10` now would be parsed as `(5 + 5) ~ 10` instead of `5 + (5 ~ 10)`. 
+
+## Effect and Interactions
+
+## Costs and Drawbacks
```

Comment:
To the costs, add that it's a breaking change, as there are programs that would be parsed differently:

```
Prelude> data a & b
Prelude> :kind! 1 ~ 2 & 3

```